azkaban-common/src/main/java/azkaban/executor/ExecutionLogsDao.java

```diff
@@ -135,12 +135,25 @@ private void uploadLogFile(final DatabaseTransOperator transOperator, final int
     }
   }
 
-  int removeExecutionLogsByTime(final long millis)
+  int removeExecutionLogsByTime(final long millis, final int recordCleanupLimit)
+      throws ExecutorManagerException {
+    int totalRecordsRemoved = 0;
+    int removedRecords;
+    do {
+      removedRecords = removeExecutionLogsBatch(millis, recordCleanupLimit);
+      logger.debug("Removed batch of execution logs. Count of records removed in this batch: "
+          + removedRecords);
+      totalRecordsRemoved = totalRecordsRemoved + removedRecords;
+    } while (removedRecords == recordCleanupLimit);
```

Comment:
Let us say we have 100K records to be deleted. Does this mean that in batch of 1000 it will run delete in while loop 100 times? Should we put upper limit as well in one run, because this thread will run every one hour? I feel it would be quite a lot of delete in one run. Also not sure how much time it will take to delete 100K records with delay in between. Just thinking about the performance and load in the DB with so many deletes at time.

Reply:
Thank you for reviewing this RB.
For clusters with huge execution_logs, I am planning to set 10K as limit. If we do so and have index on upload_time, it will take hardly 1 sec to delete 10K records. So to delete 100K records, delete query will be executed 10 times with delay of 1 sec. So it will take around 20 secs to delete all 100K records. Mysql can handle this much load so we should be good.

Reply:
Sounds good. Thanks for the clarification Janki.